### PR TITLE
Implement `Handler` for `T: IntoResponse`

### PR DESCRIPTION
**Pull request by davidpdrsn**

### CODE REVIEW

File: axum/src/handler/mod.rs [hunk 1]
```diff
@@ -224,6 +224,23 @@ macro_rules! impl_handler {
 
 all_the_tuples!(impl_handler);
 
+mod private {
+    // Marker type for `impl<T: IntoResponse> Handler for T`
+    #[allow(missing_debug_implementations)]
+    pub enum IntoResponseHandler {}
```
Comment (jplatte) [inline axum/src/handler/mod.rs:230]: I don't think this needs to have `pub` visibility? At least on my v0.6.x based branch, things compiled with a private marker type.

Comment (davidpdrsn) [reply axum/src/handler/mod.rs:230]: It didn't actually work when I tried using it

```
error: type `handler::IntoResponseHandler` is private
  --> examples/hello-world/src/main.rs:12:40
   |
12 |     let app = Router::new().route("/", get(StatusCode::OK));
   |                                        ^^^ private type
```

Comment (jplatte) [reply axum/src/handler/mod.rs:230]: What, at the usage site?? 😱

Comment (davidpdrsn) [reply axum/src/handler/mod.rs:230]: yeah that confused me as well. Haven't seen that before.

**Comment by davidpdrsn** [comment]
@jplatte I've added some docs.

**Review by jplatte — APPROVED** [review]
Awesome! 🎉 

I wouldn't mind you taking care of the backporting as well x)

**Comment by davidpdrsn** [comment]
Go for it!

**Pull request merged**